exec/cnex/array.c

```diff
@@ -45,13 +53,17 @@ Array *array_expandArray(Array *self, size_t iElements)
         if (self == NULL) {
             fatal_error("Unable to allocate array for array expansion.");
         }
-        self->size = 0;
-        self->data = NULL;
+        self->size = iElements;
```

Comment:
Under what conditions does "self" come through as NULL here? Do you need this code path? And if you do, then this block could be replaced with a call to `array_createArray()`.

Reply:
I'm not positive I need it either.  I've left the code path, but called `array_createArray()` in case this code path is ever reached.